index.rst

```diff
@@ -1,59 +1,367 @@
+
+.. warning::
+
+   NEVER use ``psql`` or other direct-SQL clients (e.g. the Python DBAPI or SQLAlchemy) to perform write operations in the repository database.
+   These can corrupt the data repository, and we have essentially no way to guard against them.
+
+   It should not be necessary in the long term to ever use direct SQL access even for read access; the SQL schema is *not* considered a public interface - but we recognize that this may be necessary for debugging for a while.
+   This can be ensured by running::
+
+      SET SESSION CHARACTERISTICS AS TRANSACTION READ ONLY;
+
+   at the start of the session.
+
+   If you have to do this (and not at the prompting of a middleware team member trying to help diagnose a problem), please also create a ticket explaining what you wanted to do that couldn't be done with butler tools, so we can address that feature gap.
+
+This proposal specifies filesystem access controls in terms of a number of high-level "roles" that certain operators or developers may temporarily opt in to via ``su`` or special setuid tools.
+Usually these roles will be used only to create subdirectories that are owned directly by the true user.
+How to map these to users, groups, and filesystem, directory, or file-level permissions in detail is something I'd prefer to leave to the system administrators.
+All directories in ``/datasets`` will be world-readable.
+
+Within each ``repo_<YYYYMMDD>`` repository directory:
+
+ - Regular users will always have write access to their own ``u/<user>`` directory.
+
+ - Production operators will have access to an ``execution`` role that can write to ``runs`` and ``<instrument>/runs`` (usually just used to create an owned subdirectory).
+
+ - Production operators and certain CPP team members will have access to a ``calibs`` role that can write to all ``<instrument>/calib`` directories (usually just used to create an owned subdirectory).
+
+ - Production operators and science pipelines developers who regularly ingest raw data for one or more instruments will have access to per-instrument ``<instrument>_raw`` roles that have write access to the ``<instrument>/raw`` and any auxiliary-data-collection subdirectories (e.g. ``HSC/masks``).  At present, these roles would have to be used directly whenever ingesting new raws, not just to create subdirectories for them, but we may be able to improve this in the future.  For external instruments, this role would also ideally provide a way to get write access (at least for writing new files) to the ``/external-raw`` filesystem (though not necessarily at that mount point).
+
+ - All production operators and science pipelines developers have access to the ``auxiliaries`` role, which provides write access to the ``refcats`` and ``skymap`` directores.  This role is used to create per-ticket subdirectories in ``refcats`` prior to starting work on ingesting a new reference catalog, and used directly to run ``butler register-skymap``.
```

Comment:
(split paragraph again?)

Per my suggestion above re:skymaps, should we restrict skymap creation to a specific `skymap` role that is only granted briefly after the "make a new skymap" RFC is successful?

Reply:
> Per my suggestion above re:skymaps, should we restrict skymap creation to a specific skymap role that is only granted briefly after the "make a new skymap" RFC is successful?

I just don't see anyone ever abusing that, in the context of a friendly, for-developers repo.

Reply:
Less a concern for abuse, and more one of accidental creation.